onmt/utils/Logger.lua

```diff
@@ -93,7 +124,61 @@ Parameters:
 
 ]]
 function Logger:error(...)
-  self:log(string.format(...), 'ERROR')
+  self:log(self:_format(...), 'ERROR')
+end
+
+--[[ Log a message at 'DEBUG' level.
+
+Parameters:
+  * `message` - the message to log. Supports formatting string.
+
+]]
+function Logger:debug(...)
+  self:log(self:_format(...), 'DEBUG')
+end
+
+--[[ Log a message as exactly it is.
+
+Parameters:
+  * `message` - the message to log. Supports formatting string.
+
+]]
+function Logger:write(...)
+  local msg = self:_format(...)
+  if (not self.mute) and self:_isVisible('WARNING') then
+    io.write(msg)
+  end
+  if self.logFile and self:_isVisible('WARNING') then
+    self.logFile:write(msg)
+    self.logFile:flush()
+  end
+end
+
+--[[ Set the visible message level. Lower level messages will be muted.
+
+Parameters:
+  * `level` - 'DEBUG', 'INFO', 'WARNING' or 'ERROR'.
+
+]]
+function Logger:setVisibleLevel(level)
```

Comment:
While you are at it, can you add a `-log_level` option so that the user can control this?